File: recyclerviewespresso/src/androidTest/java/com/allodel/recyclerviewespresso/MainActivityTest.java
package com.allodel.recyclerviewespresso;

import androidx.test.espresso.Espresso;

import androidx.test.espresso.contrib.RecyclerViewActions;
import androidx.test.rule.ActivityTestRule;
import androidx.test.runner.AndroidJUnit4;
import org.junit.Rule;
import org.junit.Test;
import org.junit.runner.RunWith;

import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.action.ViewActions.longClick;
import static androidx.test.espresso.matcher.ViewMatchers.withId;


@RunWith(AndroidJUnit4.class)
public class MainActivityTest {


    @Rule
    public ActivityTestRule<MainActivity> activityActivityTestRule = new ActivityTestRule<>(MainActivity.class);

    @Test
    public void testRecycleItemClick(){
        Espresso.onView(withId(R.id.rv_contacts)).perform(RecyclerViewActions.actionOnItemAtPosition(2,click()));
    }

    @Test
    public void testRecycleItemLongClick(){
        Espresso.onView(withId(R.id.rv_contacts)).perform(RecyclerViewActions.actionOnItemAtPosition(1,longClick()));
    }
}

File: app/src/androidTest/java/com/allodel/androiduitestingtutorials/MainActivityTest.kt
package com.allodel.androiduitestingtutorials

import androidx.test.espresso.Espresso.onView
import androidx.test.espresso.action.ViewActions.click
import androidx.test.espresso.assertion.ViewAssertions.matches
import androidx.test.espresso.matcher.ViewMatchers.withId
import androidx.test.espresso.matcher.ViewMatchers.withText
import androidx.test.rule.ActivityTestRule
 import org.junit.Test

 import org.junit.Rule

  class MainActivityTest {

      @get:Rule
      val activityTestRule = ActivityTestRule(MainActivity::class.java)


    @Test
    fun onCreate() {
      //Test views
        onView(withId(R.id.tv_quote_login))
            .check(matches(withText(R.string.quote_login)))

        onView(withId(R.id.tv_message))
            .check(matches(withText("")))

        onView(withId(R.id.et_email))
            .check(matches(withText("")))
        onView(withId(R.id.et_password))
            .check(matches(withText("")))

        onView(withId(R.id.btn_login))
            .check(matches(withText(R.string.lbl_login)))

    }

    @Test
    fun validationsuccess() {

        onView(withId(R.id.btn_login))
            .perform(click())

        onView(withId(R.id.tv_message))
            .check(matches(withText(R.string.lbl_validation_success)))



    }

}